packages/definitions-parser/src/lib/definition-parser.ts

```diff
@@ -267,7 +267,7 @@ export function getFiles(
 
   const files: string[] = [];
   function addFileIfDeclaration(path: string): void {
-    if (isDeclarationPath(path)) {
+    if (isDeclarationPath(path) || path.endsWith('/package.json') || path === 'package.json') {
```

Comment:
You made this nested function a misnomer. Either update the name or handle package.jsons separately